Assets/Scripts/UI/PauseMenu.cs

```diff
@@ -0,0 +1,37 @@
+﻿using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace UI
+{
+    public class PauseMenu : Menu
+    {
+        [SerializeField] private GameObject pausePanel;
+
+        private void Update()
+        {
+            UpdatePauseStatus();
+        }
+
+        private void UpdatePauseStatus()
+        {
+            if (Input.GetKeyDown(KeyCode.Escape))
+            {
+                Time.timeScale = 1 - Time.timeScale;
+                pausePanel.SetActive(!pausePanel.activeSelf);
+            }
```

Comment:
Big brain energy here

Reply:
I cannot take credit for this, more of @fmichaut-dev's big brain